packages/annotation-toolkit/src/layers/VideoPlayer.js

```diff
@@ -0,0 +1,128 @@
+import React, { useContext, useEffect, useCallback, useRef } from "react";
+import ReactPlayer from "react-player";
+import { useStore } from "global-context-store";
+import { Spinner } from "@blueprintjs/core";
+
+export default function VideoPlayer({
+  id,
+  src,
+  fps = 30,
+  scale,
+  width,
+  height,
+}) {
+  const store = useStore();
+  const { set, get } = store;
+  const vidRef = useRef();
+  const canvasRef = useRef();
+
+  const [detectedSize, setDetectedSize] = React.useState([10, 10]);
+  const [videoLoaded, setVideoLoaded] = React.useState(false);
+
+  width = width || detectedSize[0] * scale;
+  height = height || detectedSize[1] * scale;
+
+  const process = useCallback(
+    (req) => {
+      if (req.type === "seek") {
+        vidRef.current.seekTo(req.payload, "seconds");
+      } else if (req.type === "rewind") {
+        const currentTime = get(path("playedSeconds"));
+        vidRef.current.seekTo(
+          Math.max(currentTime - req.payload, 0),
+          "seconds"
+        );
+      } else if (req.type === "ff") {
+        const currentTime = get(path("playedSeconds"));
+        const duration = get(path("duration"));
+        vidRef.current.seekTo(
+          Math.min(currentTime + req.payload, duration),
+          "seconds"
+        );
+      } else if (req.type === "screenshot") {
+        canvasRef.current
+          .getContext("2d")
+          .drawImage(vidRef.current.getInternalPlayer(), 0, 0, width, height);
+        const screenshotData = canvasRef.current.toDataURL("image/png");
+        req.payload.callback({ store, data: screenshotData });
+      }
+    },
+    [vidRef.current]
+  );
+
+  React.useEffect(() => {
+    /* Determine video dimensions */
+    if (!videoLoaded) return;
+    if (!vidRef.current.getInternalPlayer()) return;
+    const videoWidth = vidRef.current.getInternalPlayer().videoWidth;
+    const videoHeight = vidRef.current.getInternalPlayer().videoHeight;
+    set(path("detectedSize"), [videoWidth, videoHeight]);
```

Comment:
Does it make sense to extract variables as constants? Help avoid bugs due to typos.

Reply:
This is being tracked with #433 will likely add in the future